README.md

```diff
@@ -0,0 +1,93 @@
+# httpio
+
+The `httpio` package provides tools for encoding HTTP responses and decoding HTTP requests in Go, complete with validation rules.
+
+## Getting Started
+
+First, get the package by running:
+
+```sh
+go get github.com/cccteam/httpio
+```
+
+## Decoder
+
+The Decoder struct is used to decode and validate HTTP requests. It utilizes the json.NewDecoder() function to decode the HTTP request body into a provided struct.
+
+Validation is handled by the Validator interface, which requires a Struct(s interface{}) error function. This function is expected to validate the struct s and return an error if the validation fails.
+
+### Example usage
+
+```go
+type MyValidator struct{}
+
+func (v MyValidator) Struct(s interface{}) error {
+    // validation logic goes here
+}
+
+type MyRequest struct {
+    Field1 string `json:"field1" validate:"required"`
+    Field2 int    `json:"field2" validate:"required,gt=0"`
+}
+
+func MyHandler(w http.ResponseWriter, r *http.Request) {
+    req := &MyRequest{}
+    decoder := httpio.NewDecoder(r, MyValidator{})
+    if err := decoder.Decode(req); err != nil {
+        // handle error
+        return
+    }
+    // continue processing the request...
+}
+```

```

Comment:
This example no longer matches the change to the `Validator` interface.

Reply:
Oops